{"context": "WP_UnitTestCase\u3092\u5229\u7528\u3057\u3066\u3044\u308b\u5834\u5408\u3001WP_Error\u30aa\u30d6\u30b8\u30a7\u30af\u30c8\u3092\u6b63\u3057\u304f\u8fd4\u3057\u3066\u5c45\u308b\u304b\u5426\u304b\u306e\u30c6\u30b9\u30c8\u306bassertWPError\u304c\u5229\u7528\u3067\u304d\u307e\u3059\u3002\n        function test_is_wp_error() {\n                $test = new WP_Error( 'something happen' );\n                $this->assertWPError( $test );\n        }\n\n\u3061\u306a\u307f\u306bassertNotWPError\u3092\u4f7f\u3046\u3068\u3001\u300cWP_Error\u3067\u306f\u306a\u3044\u3053\u3068\u300d\u3092\u30c6\u30b9\u30c8\u3067\u304d\u307e\u3059\u3002\n    function test_is_not_wp_error() {\n        $test = 'nothing happen';\n        $this->assertNotWPError( $test );\n    }\n\n\u3053\u3061\u3089\u306fWordPress4.5.4\u3060\u3068PHP Fatal error:  Call to undefined method\u304c\u51fa\u307e\u3057\u305f\u306e\u3067\u3001\u5229\u7528\u306b\u306f\u8981\u6ce8\u610f\u3067\u3059\u3002\n\n\u304a\u307e\u3051\uff1a\u4e21\u95a2\u6570\u306e\u30bd\u30fc\u30b9\u30b3\u30fc\u30c9\n\u30bd\u30fc\u30b9\u30b3\u30fc\u30c9\uff1ahttp://develop.svn.wordpress.org/trunk/tests/phpunit/includes/testcase.php\n    function assertWPError( $actual, $message = '' ) {\n        $this->assertInstanceOf( 'WP_Error', $actual, $message );\n    }\n\n    function assertNotWPError( $actual, $message = '' ) {\n        if ( is_wp_error( $actual ) && '' === $message ) {\n            $message = $actual->get_error_message();\n        }\n        $this->assertNotInstanceOf( 'WP_Error', $actual, $message );\n    }\n\nPHPUnit\u306eassertInstanceOf()\u3068\u3001assertNotInstanceOf()\u306e\u30e9\u30c3\u30d1\u30fc\u306b\u306a\u3063\u3066\u3044\u307e\u3059\u3002\n\n\n`WP_UnitTestCase`\u3092\u5229\u7528\u3057\u3066\u3044\u308b\u5834\u5408\u3001`WP_Error`\u30aa\u30d6\u30b8\u30a7\u30af\u30c8\u3092\u6b63\u3057\u304f\u8fd4\u3057\u3066\u5c45\u308b\u304b\u5426\u304b\u306e\u30c6\u30b9\u30c8\u306b`assertWPError`\u304c\u5229\u7528\u3067\u304d\u307e\u3059\u3002\n\n\n```php\n        function test_is_wp_error() {\n                $test = new WP_Error( 'something happen' );\n                $this->assertWPError( $test );\n        }\n```\n\n\n\u3061\u306a\u307f\u306b`assertNotWPError`\u3092\u4f7f\u3046\u3068\u3001\u300cWP_Error\u3067\u306f\u306a\u3044\u3053\u3068\u300d\u3092\u30c6\u30b9\u30c8\u3067\u304d\u307e\u3059\u3002\n\n```php\n\tfunction test_is_not_wp_error() {\n\t\t$test = 'nothing happen';\n\t\t$this->assertNotWPError( $test );\n\t}\n```\n\n\u3053\u3061\u3089\u306fWordPress4.5.4\u3060\u3068`PHP Fatal error:  Call to undefined method`\u304c\u51fa\u307e\u3057\u305f\u306e\u3067\u3001\u5229\u7528\u306b\u306f\u8981\u6ce8\u610f\u3067\u3059\u3002\n\n## \u304a\u307e\u3051\uff1a\u4e21\u95a2\u6570\u306e\u30bd\u30fc\u30b9\u30b3\u30fc\u30c9\n\n\u30bd\u30fc\u30b9\u30b3\u30fc\u30c9\uff1ahttp://develop.svn.wordpress.org/trunk/tests/phpunit/includes/testcase.php\n\n```php\n\tfunction assertWPError( $actual, $message = '' ) {\n\t\t$this->assertInstanceOf( 'WP_Error', $actual, $message );\n\t}\n\n\tfunction assertNotWPError( $actual, $message = '' ) {\n\t\tif ( is_wp_error( $actual ) && '' === $message ) {\n\t\t\t$message = $actual->get_error_message();\n\t\t}\n\t\t$this->assertNotInstanceOf( 'WP_Error', $actual, $message );\n\t}\n```\n\nPHPUnit\u306e[assertInstanceOf()\u3068\u3001assertNotInstanceOf()](https://phpunit.de/manual/current/ja/appendixes.assertions.html#appendixes.assertions.assertInstanceOf)\u306e\u30e9\u30c3\u30d1\u30fc\u306b\u306a\u3063\u3066\u3044\u307e\u3059\u3002\n", "tags": ["WP_UnitTestCase", "WordPress", "PHPUnit"]}